src/DiyaNode.js

```
@@ -563,13 +586,15 @@ DiyaNode.prototype._handlePeerDisconnected = function(message){
 };
 
 DiyaNode.prototype._handleRequest = function(handler, message){
+
 	if(message.type === 'PartialAnswer') {
 		if(typeof this._pendingMessages[message.id].callback_partial === 'function') {
 			var error = message.error ? message.error : null;
 			var data = message.data ? message.data : null;
 			this._pendingMessages[message.id].callback_partial(error, data);
 		}
```

else ?

What happens if `typeof ...` is not a `function`? You should return an error, no?